### PR TITLE
[13.0][FIX] dms: Avoid error when accessing a subdirectory without access token

### PR DESCRIPTION
**Pull request by victoralmau**
Misc changes:
- [x] Avoid error when accessing a subdirectory without access token
- [x] Use the correct domain with inherit_access_from_parent_record field

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa

### CODE REVIEW

File: dms/models/directory.py [hunk 1]
```diff
@@ -260,7 +260,11 @@ def _get_parent_categories(self, access_token):
         current_directory = self
         while current_directory:
             directories.insert(0, current_directory)
-            if access_token and consteq(current_directory.access_token, access_token):
+            if (
```
Comment (pedrobaeza) [inline dms/models/directory.py:263]: But is this taking into account subdirectories of a shared directory?

Comment (victoralmau) [reply dms/models/directory.py:263]: Of course, if you share a directory, you will have access to subdirectories (and the breadcrumb will only show the directories from which you have access).

Comment (pedrobaeza) [reply dms/models/directory.py:263]: I mean the following scenario:

- You share via URL with token a directory.
- That directory has subdirectories that are seen on the portal.
- You click on a subdirectory.

Do you see the contents of such subdirectory?

Comment (victoralmau) [reply dms/models/directory.py:263]: Yes, a small example (similar to any other system, if you share something you share EVERYTHING in it).

dms

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Pull request merged**